### ISSUE TITLE
Root URLs cannot be on the web?

### ISSUE THREAD

**Issue by tintori** [ptgolden/dredge#16]
I uploaded all my data files to my personal website and tried setting that as my root directory (as pictured). Even though all the files/settings are exactly the same is my local version, when I hit test it arrests immediately (as pictured).
Ultimately the plan is to allow people to host data online and share projects publically,right—can you explain how this will fit in to what we already have?

**Comment by tintori** [comment]
Whoops, here are the images.

Settings in "Configuration":
![screen shot 2019-03-04 at 9 23 30 pm](https://user-images.githubusercontent.com/7449762/53776501-3f6b0080-3ec4-11e9-8f6c-91b69df265a9.png)
Error message:
![screen shot 2019-03-04 at 9 24 01 pm](https://user-images.githubusercontent.com/7449762/53776533-51e53a00-3ec4-11e9-8447-82b4169d7c23.png)

When I click on the resolved URL for this file, it loads without a problem.

**Comment by tintori** [comment]
More info:
If I put the whole package together locally and then upload it to my personal website, though, it works just fine.
http://sophiatintori.com/holding/dredge-4.2.1-wormtest/?page=home&project=worm%20developmental%20timecourse

**Comment by ptgolden** [comment]
This is an issue with your Web server. By default, a web page is not allowed to make requests to another domain. (This is a fundamental part of Web security that we can't change).

You need to configure your server to use [CORS headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) when serving those requests. I'm not sure where your server is hosted or how it's configured, but I'd be happy to help you figure out how you can do that.

Basically, if something will be requested from another domain, they must have this header set in the response:
```
Access-Control-Allow-Origin: *
```

**Comment by ptgolden** [comment]
Adding, the reason it works if you upload the configuration file to your Web server, it works because it's requesting resources from the same domain.

While this is confusing-- and should/will be documented better-- it's fundamentally how the data loading process works. What I imagined the situation would be is that you would develop a new site on your own machine, keeping everything with relative URLs, and then moving all of those files onto a server once the configuration works.

**Comment by tintori** [comment]
Yep, ok that sounds good. So then our challenge is to make the documentation clearer, and organize the users' flow through the website so that they're not tempted to start developing the new site before moving everything to their own machine—both of which are things we're already planning to do. hmm lets see howd you do that thumbs up ... 👍

**Comment by ptgolden** [comment]
Is this better at all?

http://dredge.bio.unc.edu/c-elegans-timecourse/?page=configure

**Comment by ptgolden** [comment]
I made it match the instructions, and also added this:

> The directory where you will save the configuration file you generate from this page, relative to where you will place the index.html file when you upload your DrEdGE project. All of the following settings involving URLs will be resolved relative to this directory.
>
> If you change this URL, it can be relative or absolute, But note that if data will be served from a different host than the DrEdGE page, you will need to set appropriate CORS headers on that host.

**Comment by tintori** [comment]
Slight tweak that might make the "relative" part clearer:

> The path from the directory where the DrEdGE index.html file will live to the directory where you will save the configuration file generated from this form. All the following settings involving....

That's accurate, right?

**Comment by ptgolden** [comment]
Yeah, I agree, this could be clearer. (These things are hard to write clearly).

I understand what you mean, but we should keep "relative" in. It's not common to talk about a path "from" somewhere, the common way of expressing that concept is "relative to". Someone who is familiar with the command line will very likely be familiar with the terminology, and that is who I imagine we are talking to with these instructions.

To clarify, based on your suggestion, how about...

> The path to the directory where you will save the configuration file generated from this form, relative to the directory where the DrEdGE index.html will live.

**Comment by tintori** [comment]
Yep, sounds good. 

I agree, it's hard to write this stuff clearly, and there are two types of technical language to balance at the same time—perhaps we can spend half an hour one day soon just combing through all the text together, to make sure it all feels clear to both of us.
